src/Modal/Modal.js

```diff
@@ -98,8 +108,27 @@ function Modal({ children, title, animationDuration, showClose, onClose, open, .
     if (open !== isOpen) {
       setOpen(open);
     }
+
+    toggleBlur();
   }, [open]);
 
+  useEffect(() => {
+    if (blur) {
+      const target = document.querySelector(blurTarget);
+
+      if (target) {
+        blurNode.current = target;
+
+        blurNode.current.style.overflow = 'auto';
+
+        return () => {
+          blurNode.current.style.filter = null;
+          blurNode.current.style.overflow = null;
```

Comment:
How about keeping track of the initial values for `overflow` and `filter`, so that they can be restored here and when the filter is being set?